incubator/group/types.proto

```diff
@@ -304,3 +304,82 @@ message GenesisState {
     option (gogoproto.goproto_stringer) = false;
     Params Params = 1 [(gogoproto.nullable)=false];
 }
+
+//
+// Queries
+//
+service Query {
+    rpc GetGroupMetadata(GroupMetadataRequest) returns (GroupMetadata);
+    rpc GetGroupAccountMetadata(GroupAccountMetadataRequest) returns (AnyGroupAccountMetadata);
+    rpc GetGroupsByMember(ByAddressRequest) returns (GroupsList);
+    rpc GetGroupsByOwner(ByAddressRequest) returns (GroupsList);
+    rpc GetGroupAccountsByGroup(ByGroupRequest) returns (GroupAccountsList);
+    rpc GetGroupAccountsByOwner(ByAddressRequest) returns (GroupAccountsList);
+    rpc GetProposal(ProposalRequest) returns (AnyProposal);
```

Comment:
Some entities may not contain their own ID. While we could return them as they are it would be impossible for a client to do any useful operation with them. For example `Vote` requires a `Proposal ID` which is not part of `ProposalBase`.
In Weave we return the result as a list of Model{Key,Value} entries.